patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -511,6 +511,23 @@ private class HookList
 			return true;
 		}
 
+		private delegate bool? DelegateConsumeBait(Item bait, Player player);
+		private static HookList HookConsumeBait = AddHook<DelegateConsumeBait>(p => p.ConsumeBait);
+		public static bool? ConsumeBait(Player player, Item bait) {
+			if (bait.IsAir)
+				return null;
+
+			bool? ans = bait.modItem?.ConsumeBait(player);
+			if (ans != null)
+				return ans;
+			foreach (var g in HookConsumeBait.arr) {
+				ans = g.ConsumeBait(bait, player);
+				if (ans != null)
+					return ans;
+			}
+			return null;
```

Comment:
Global hooks should take priority over mod ones. In other words, loop through the global hooks first, then run the mod hook.

Reply:
Copy this: https://github.com/tModLoader/tModLoader/blob/21166e09b6d0c526957f3e50cc8f01d6de6c5bc7/patches/tModLoader/Terraria/ModLoader/ProjectileLoader.cs#L300